fuse/mount_darwin.go

```diff
@@ -14,33 +14,106 @@ import (
 	"unsafe"
 )
 
+const FUSET_SRV_PATH = "/usr/local/bin/go-nfsv4"
+
+var osxFuse bool
+
 func unixgramSocketpair() (l, r *os.File, err error) {
 	fd, err := syscall.Socketpair(syscall.AF_UNIX, syscall.SOCK_STREAM, 0)
 	if err != nil {
 		return nil, nil, os.NewSyscallError("socketpair",
 			err.(syscall.Errno))
 	}
-	l = os.NewFile(uintptr(fd[0]), "socketpair-half1")
-	r = os.NewFile(uintptr(fd[1]), "socketpair-half2")
+	l = os.NewFile(uintptr(fd[0]), fmt.Sprintf("socketpair-half%d", fd[0]))
+	r = os.NewFile(uintptr(fd[1]), fmt.Sprintf("socketpair-half%d", fd[1]))
 	return
 }
 
+func mount(mountPoint string, opts *MountOptions, ready chan<- error) (fd int, err error) {
+	if fuset_bin, err := fusetBinary(); err == nil {
+		osxFuse = false
+		return mount_fuset(fuset_bin, mountPoint, opts, ready)
+	} else if osxfuse_bin, err := fusermountBinary(); err == nil {
+		osxFuse = true
+		return mount_osxfuse(osxfuse_bin, mountPoint, opts, ready)
+	}
+	return -1, fmt.Errorf("not FUSE-T nor osxFuse found")
+}
+
 // Create a FUSE FS on the specified mount point.  The returned
 // mount point is always absolute.
-func mount(mountPoint string, opts *MountOptions, ready chan<- error) (fd int, err error) {
-	local, remote, err := unixgramSocketpair()
+var local, local_mon, remote, remote_mon *os.File
```

Comment:
Unfortunately using local  variables in this context would not work because they'd become garbage collected. The reason is that the function doesn't return *os.File for send/receive but the underlying file descriptor and GC doesn't know about that. I will provide an explanatory comment